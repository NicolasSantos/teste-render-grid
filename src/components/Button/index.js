import React from 'react';
import './index.scss';

const Button = ({icon, onClick}) => {
    return (
        <button onClick={() => onClick()}>
            {icon}
        </button>
    )
}

export default Button;