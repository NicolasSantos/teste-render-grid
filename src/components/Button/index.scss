@import '../../styles/abstracts/variables.scss';

button {
    height: 40px;
    padding: 5px 15px;
    background: $primary-color;
    color: $secondary-color;
    font-size: 17px;
    cursor: pointer;
    box-sizing: border-box;
    border: 1px solid $button-border;
    border-left: none;

    &:hover {
        opacity: 0.8;
    }
}