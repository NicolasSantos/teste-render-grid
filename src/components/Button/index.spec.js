import React from "react";
import { shallow } from "enzyme";
import Button from "./index";

describe("[COMPONENT] - Button", () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<Button/>);
    });

    it('Snapshot Button', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('Render button with icon', () => {
        const iconElement = <i>Icon</i>
    
        expect(wrapper.contains(iconElement)).toBeFalsy();

        wrapper.setProps({ icon: iconElement });

        expect(wrapper.contains(iconElement)).toBeTruthy();
    });

    it('Click button', () => {
        const mockCallback = jest.fn();
    
        const button = shallow((<Button onClick={mockCallback}/>));
        button.find('button').simulate('click');
        
        expect(mockCallback.mock.calls.length).toEqual(1);
    });
})