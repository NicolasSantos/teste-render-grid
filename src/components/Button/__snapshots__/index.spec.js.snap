// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[COMPONENT] - Button Snapshot Button 1`] = `
<button
  onClick={[Function]}
/>
`;
